chain/xrp/client/client.go

```diff
@@ -0,0 +1,816 @@
+				sources = append(sources, &xc.LegacyTxInfoEndpoint{
+					Address: xc.Address(txResponse.Result.Account),
+					Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerGets.XRPAmount),
+				})
+
+				for _, node := range txResponse.Result.Meta.AffectedNodes {
+					if node.ModifiedNode != nil {
+						if node.ModifiedNode.LedgerEntryType == "AccountRoot" {
+							var (
+								finalBalance, previousBalance int64
+								parseIntErr                   error
+							)
+
+							if node.ModifiedNode.FinalFields != nil {
+								finalBalance, parseIntErr = strconv.ParseInt(node.ModifiedNode.FinalFields.Balance.XRPAmount, 10, 64)
+								if parseIntErr != nil {
+									return xc.LegacyTxInfo{}, parseIntErr
+								}
+							} else {
+								break
+							}
+
+							if node.ModifiedNode.PreviousFields != nil {
+								previousBalance, parseIntErr = strconv.ParseInt(node.ModifiedNode.PreviousFields.Balance.XRPAmount, 10, 64)
+								if parseIntErr != nil {
+									return xc.LegacyTxInfo{}, parseIntErr
+								}
+							} else {
+								break
+							}
+
+							transactedAmount := finalBalance - previousBalance
+
+							soldAmount, parseIntErr := strconv.ParseInt(txResponse.Result.TakerGets.XRPAmount, 10, 64)
+							if parseIntErr != nil {
+								return xc.LegacyTxInfo{}, parseIntErr
+							}
+
+							if transactedAmount == soldAmount {
+								destinations = append(destinations, &xc.LegacyTxInfoEndpoint{
+									Address: xc.Address(node.ModifiedNode.FinalFields.Account),
+									Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerGets.XRPAmount),
+								})
+							}
+
+						}
+					}
+				}
+			} else {
+				amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerGets.TokenAmount.Value)
+				if err != nil {
+					return xc.LegacyTxInfo{}, err
+				}
+				sources = append(sources, &xc.LegacyTxInfoEndpoint{
+					Address: xc.Address(txResponse.Result.Account),
+					Amount:  amount.ToBlockchain(15),
+					Asset:   txResponse.Result.TakerGets.TokenAmount.Currency,
+				})
+			}
+		}
+
+		if txResponse.Result.TakerPays != nil {
+			if txResponse.Result.TakerPays.XRPAmount != "" {
+				destinations = append(destinations, &xc.LegacyTxInfoEndpoint{
+					Address: xc.Address(txResponse.Result.Account),
+					Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerPays.XRPAmount),
+				})
+			} else {
+				amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerPays.TokenAmount.Value)
+				if err != nil {
+					return xc.LegacyTxInfo{}, err
+				}
+
+				sources = append(sources, &xc.LegacyTxInfoEndpoint{
+					Address: xc.Address(txResponse.Result.TakerPays.TokenAmount.Issuer),
+					Amount:  amount.ToBlockchain(15),
+					Asset:   txResponse.Result.TakerPays.TokenAmount.Currency,
+				})
+
+				for _, node := range txResponse.Result.Meta.AffectedNodes {
+					if node.ModifiedNode != nil {
+						if node.ModifiedNode.LedgerEntryType == "RippleState" {
```

Comment:
you could simplify this a bit by defining a helper method on the type of `txResponse.Result.Meta`.

```go
func (m *Meta) GetAffectedNodesByType(entryType string) []*AffectedNode {
    nodes := []*AffectedNode{}
    for _, node := range txResponse.Result.Meta.AffectedNodes {
	    if node.ModifiedNode != nil {
		    if node.ModifiedNode.LedgerEntryType == entryType {
                        nodes = append(nodes, node)
                    }
            }
    }

    return nodes
}

// later
for _, node := range txResponse.Result.Meta.GetAffectedNodesByType("RippleState") {
```